src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -81,141 +76,73 @@ public ExamAccessService(ExamRepository examRepository, ExerciseGroupRepository
+                break;
+            case STUDENT:
+                authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
+                break;
+            default:
+                break;
         }
-        return Optional.empty();
     }
 
     /**
-     * Checks if the current user is allowed to access the exam as teaching assistant.
+     * Checks if the current user is allowed to manage exams of the given course, otherwise throws an AccessForbiddenException
      *
+     * @param role the role to check the access for in the course
      * @param courseId The id of the course
-     * @param <T>      The type of the return type of the requesting route so that the response can be returned there
-     * @return an optional with a typed ResponseEntity. If it is empty all checks passed
+     * @param examId the id of the exam
+     * @throws AccessForbiddenException if user isn't at least of role in course
      */
-    public <T> Optional<ResponseEntity<T>> checkCourseAccessForTeachingAssistant(Long courseId) {
-        Course course = courseRepository.findByIdElseThrow(courseId);
-        if (!authorizationCheckService.isAtLeastTeachingAssistantInCourse(course, null)) {
-            return Optional.of(forbidden());
-        }
-        return Optional.empty();
+    public void checkCourseAndExamAccessForRoleElseThrow(Role role, Long courseId, Long examId) {
+        Exam exam = examRepository.findByIdElseThrow(examId);
+        checkCourseAndExamAccessForRoleElseThrow(role, courseId, exam);
     }
 
-    /**
-     * Checks if the current user is allowed to manage exams of the given course, that the exam exists and that the exam
-     * belongs to the given course.
-     *
-     * @param courseId The id of the course
-     * @param examId   The id of the exam
-     * @param <X>      The type of the return type of the requesting route so that the response can be returned there
-     * @return an optional with a typed ResponseEntity. If it is empty all checks passed
-     */
-    public <X> Optional<ResponseEntity<X>> checkCourseAndExamAccessForEditor(Long courseId, Long examId) {
-        Optional<ResponseEntity<X>> courseAccessFailure = checkCourseAccessForEditor(courseId);
-        if (courseAccessFailure.isPresent()) {
-            return courseAccessFailure;
-        }
-        return checkCourseAndExamAccess(courseId, examId);
+    public void checkCourseAndExamAccessForRoleElseThrow(Role role, Long examId) {
+        Exam exam = examRepository.findByIdElseThrow(examId);
+        checkCourseAndExamAccessForRoleElseThrow(role, exam.getCourse().getId(), exam);
     }
 
     /**
-     * Checks if the current user is allowed to manage exams of the given course, that the exam exists and that the exam
-     * belongs to the given course.
+     * Checks if the current user is allowed to manage exams of the given course, otherwise throws an AccessForbiddenException
```

Comment:
I think either the documentation or the implementation is wrong/misleading: If the method is called with `Roll.STUDENT` it will be forwarded to `checkCourseAccessForRoleElseThrow(STUDENT, id)` which forwards it to `authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);`. This succeeds if the student is part of the course. However, a student is not allowed to manage exams.